src/mmrelay/matrix/sync_bootstrap.py

```
@@ -0,0 +1,1201 @@
+            )
+
+            if credentials_path:
+                path_exists = await asyncio.to_thread(os.path.exists, credentials_path)
+            else:
+                path_exists = False
+            existing_credentials_path = credentials_path if path_exists else None
+            if existing_credentials_path is not None:
+
+                def _load_direct(path: str) -> dict[str, Any] | None:
+                    try:
+                        with open(path, "r", encoding="utf-8") as f:
```

Ruff (0.15.9)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)